src/Hl7.Fhir.Specification/Specification/Snapshot/SnapshotGenerator.cs

```diff
@@ -890,6 +884,11 @@ private async T.Task mergeElement(ElementDefinitionNavigator snap, ElementDefini
                 // Now, recursively merge the children
                 await merge(snap, diff).ConfigureAwait(false);
 
+                // [MS 20220425] Make sure newly added contentReferences (from bases, types, and contentReferences) are absolute.
+                // Except when the it's a core profile (Derivation is not a Constraint)
+                if (snap.StructureDefinition.Derivation == StructureDefinition.TypeDerivationRule.Constraint)
+                    ensureAbsoluteContentReferences(snap, snap.StructureDefinition.BaseDefinition);
```

Comment:
Why cannot we call `ensureAbsoluteContentReference()` with `defn.ContentReferenceElement` here directly?  Why using the navigate to the first child first? I think `defn` is already the right child here?